react-delegationdashboard/src/components/Overview/Cards/index.tsx

```diff
@@ -0,0 +1,190 @@
+import { ContractOverview, NodeType } from 'helpers/types';
+import { Query, QueryResponse } from '@elrondnetwork/erdjs/out/smartcontracts/query';
+import { Address, Argument, ContractFunction, NetworkStake } from '@elrondnetwork/erdjs/out';
+import { useState } from 'react';
+
+import SetPercentageFeeAction from './SetPercentageFeeAction';
+import UpdateDelegationCapAction from './UpdateDelegationCapAction';
+import AutomaticActivationAction from './AutomaticActivationAction';
+
+const Views = () => {
+  const { dapp, egldLabel, delegationContract, auctionContract, address } = useContext();
+  const { getTotalActiveStake, getContractConfig } = contractViews;
+  const [totalActiveStake, setTotalActiveStake] = React.useState('...');
+  const [noNodes, setNoNodes] = React.useState('...');
+  const [contractOverview, setContractOverview] = useState(new ContractOverview());
+  const [networkStake, setNetworkStake] = useState(new NetworkStake());
+  const [isAdminFlag, setIsAdminFlag] = useState(false);
+
+  const getContractOverviewType = (value: QueryResponse) => {
+    let delegationCap = denominate({
+      decimals,
+      denomination,
+      input: value.returnData[2].asBigInt.toString(),
+      showLastNonZeroDecimal: false,
+    });
+    let initialOwnerFunds = denominate({
+      decimals,
+      denomination,
+      input: value.returnData[3].asBigInt.toString(),
+      showLastNonZeroDecimal: false,
+    });
+    return new ContractOverview(
+      (parseFloat(value.returnData[1].asHex) / 100).toString(),
+      delegationCap,
+      initialOwnerFunds,
+      value.returnData[4]?.asString,
+      value.returnData[5].asBool,
+      value.returnData[6].asBool,
+      value.returnData[7].asBool,
+      value.returnData[8]?.asNumber * 6
+    );
+  };
+
+  const getPercentage = (firstValue: string, secondValue: string) => {
+    let percentage =
+      (parseInt(firstValue.replace(/,/g, '')) / parseInt(secondValue.replace(/,/g, ''))) * 100;
+    if (percentage < 1) {
+      return '<1';
+    }
+    return percentage ? percentage.toFixed(2) : '...';
+  };
+
+  const isAdmin = (ownerAddress: string) => {
+    let loginAddress = new Address(address).hex();
+    return loginAddress === ownerAddress;
+  };
+
+  const getContractConfiguration = () => {
+    getContractConfig(dapp, delegationContract)
+      .then(value => {
+        if (isAdmin(value.returnData[0].asHex)) {
+          setIsAdminFlag(true);
+        }
+        let contractOverview = getContractOverviewType(value);
+        setContractOverview(contractOverview);
+      })
+      .catch(e => console.error('getContractConfig error ', e));
+  };
+
+  const getNumberOfActiveNodes = () => {
+    const query = new Query({
+      address: new Address(auctionContract),
+      func: new ContractFunction('getBlsKeysStatus'),
+      args: [Argument.fromPubkey(new Address(delegationContract))],
+    });
+    return new Promise<Array<NodeType>>(resolve => {
```

Comment:
We don't need to return another promise here. The top one is never actually resolved, and in the only place `getNumberOfActiveNodes` is called, the promise is not handled. So the approach in the functions above seems good enough

Reply:
done